cmd/algokey/keyreg.go

```diff
@@ -0,0 +1,264 @@
+// Copyright (C) 2019-2022 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package main
+
+import (
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"github.com/algorand/go-algorand/crypto"
+	"github.com/algorand/go-algorand/data/account"
+	"github.com/algorand/go-algorand/data/basics"
+	"github.com/algorand/go-algorand/data/transactions"
+	"github.com/algorand/go-algorand/protocol"
+	"github.com/algorand/go-algorand/util"
+	"github.com/algorand/go-algorand/util/db"
+)
+
+var keyregCmd *cobra.Command
+
+type keyregCmdParams struct {
+	fee         uint64
+	firstValid  uint64
+	lastValid   uint64
+	network     string
+	offline     bool
+	txFile      string
+	partkeyFile string
+	addr        string
+}
+
+// There is no node to query, so we do our best here.
+const (
+	txnLife uint64 = 1000
+	minFee  uint64 = 1000
+)
+
+type networkGenesis struct {
+	id   string
+	hash crypto.Digest
+}
+
+var validNetworks map[string]networkGenesis
+var validNetworkList []string
+
+func init() {
+	var params keyregCmdParams
+
+	keyregCmd = &cobra.Command{
+		Use:   "keyreg",
+		Short: "Make key registration transaction",
+		Args:  cobra.NoArgs,
+		Run: func(cmd *cobra.Command, _ []string) {
+			err := run(params)
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "%s\n\n", err.Error())
+				os.Exit(1)
+			}
+		},
+	}
+
+	keyregCmd.Flags().Uint64Var(&params.fee, "fee", minFee, "transaction fee")
+	keyregCmd.Flags().Uint64Var(&params.firstValid, "first-valid", 0, "first round where the transaction may be committed to the ledger")
```

Comment:
please use the same parameter names ( when possible ), as in goal to retain some level of consistency. In this case `first-valid` -> `firstvalid`

Reply:
Thanks. It looks like there are 2 problems with this.
1) algokey uses `first` and `last` to define the key ranges. Technically this is slightly different, being the transaction ranges, so I'll keep consistent with goal on that one.
2) algokey refers to the key file as `keyfile` and `goal account installpartkey` refers to it as `partkey`. For this I'll keep consistent with algokey.